lib/BridgeAbstract.php

```diff
@@ -298,4 +298,36 @@ public function detectParameters($url){
 			return null;
 		}
 	}
+
+	/**
+	 * Loads a cached value for the specified key
+	 *
+	 * @param string $key Key name
+	 * @param int $duration Cache duration (optional, default: 24 hours)
+	 * @return mixed Cached value or null if the key doesn't exist or has expired
+	 */
+	protected function loadCacheValue($key, $duration = 86400){
+		$cacheFac = new CacheFactory();
+		$cacheFac->setWorkingDir(PATH_LIB_CACHES);
+		$cache = $cacheFac->create(Configuration::getConfig('cache', 'type'));
+		$cache->setScope(get_called_class());
+		$cache->setKey($key);
+		$cache->purgeCache($duration);
```

Comment:
Just noticed that this operation will purge the entire cache for the bridge despite the fact that some keys might have a longer duration.

Reply:
Oh you're right. I suppose that's what `purgeCache()` is intended to do. Maybe either a `deleteKey()` function could be added to the `CacheInterface`, or have `purgeCache()` redefined to only remove a single key if one is set, but that sounds like it could cause other problems.

I suppose another option is to just check `$cache->getTime()` and return null if the key is older than `$duration`, without removing the cache entry at all. That might be easiest.

Reply:
> I suppose another option is to just check `$cache->getTime()` and return null if the key is older than `$duration`, without removing the cache entry at all. That might be easiest.

I think so too. In fact, RSS-Bridge does a global purge after 24 hours anyway.